src/base/Notifier.sol

```diff
@@ -30,26 +28,24 @@ abstract contract Notifier is INotifier {
     mapping(uint256 tokenId => ISubscriber subscriber) public subscriber;
 
     modifier onlyIfApproved(address caller, uint256 tokenId) virtual;
-    modifier onlyValidConfig(uint256 tokenId, PositionConfig calldata config) virtual;
 
-    function _positionConfigs(uint256 tokenId) internal view virtual returns (PositionConfigId storage);
+    function _setUnsubscribe(uint256 tokenId) internal virtual;
+
+    function _setSubscribe(uint256 tokenId) internal virtual;
 
     /// @inheritdoc INotifier
-    function subscribe(uint256 tokenId, PositionConfig calldata config, address newSubscriber, bytes calldata data)
+    function subscribe(uint256 tokenId, address newSubscriber, bytes calldata data)
         external
         payable
         onlyIfApproved(msg.sender, tokenId)
-        onlyValidConfig(tokenId, config)
     {
-        // will revert below if the user already has a subcriber
-        _positionConfigs(tokenId).setSubscribe();
+        _setSubscribe(tokenId);
```

Comment:
i think to officially follow checks-effects-interactions as our audits have brought up this should go after the check that
```
if (_subscriber != NO_SUBSCRIBER)
```

so it does that check, then sets subscribers in both ways, and then does the call